### ISSUE TITLE
Exercism upgrade not working

### ISSUE THREAD

**Issue by joce** [exercism/cli#391]
Running `exercism upgrade` gives me the following:

<pre>
Downloading exercism-windows-64bit.zip
2017/05/11 13:31:14 open C:\ProgramData\chocolatey\lib\exercism-io-cli\tools\exercism.exe: The process cannot access the file because it is being used by another process.
</pre>

Also, why is the site recommending to use `exercism upgrade` while the install guide gives steps to install using choco? And finally, why is the version on choco 2.3.0 while the version advertised on the site is 2.4.0?

**Comment by Tonkpils** [comment]
I have a PR out that will hopefully fix the issue with the upgrade command on Windows machines. 

I'm not sure who is the maintainer for the `choco` installer but I'm assuming that has been left behind a bit. Our current CLI version is `2.4.0`. If you're having issues upgrading, you can fetch the latest version from https://github.com/exercism/cli/releases and replace the old executable you have with the one you download.

**Comment by joce** [comment]
`choco` is still the way listed on the web site as to how to install on Windows: http://exercism.io/clients/cli/windows 
Should I open a separate issue to ask for an update to the `choco` installer?

**Comment by kytrinyx** [comment]
@joce Installing the Exercism command-line client for the first time is probably easier using the Choco installer, but once you have it installed, I'd recommend using the `upgrade` command (provided that it works).

**Comment by joce** [comment]
but then choco gets out of sync.

running `choco outdated` will show a version that isn't the actual one.

**Comment by NobbZ** [comment]
Since this came up again in the gitter.im/exercism/support I did some research and filed a bug at Spinarooni/chocolatey-packages#1

**Comment by NobbZ** [comment]
If there is no response by @Spinarooni in a reasonable amount of time, I think we should either create a choco package on our own, owned by the exercism organisation (or Katrina) or remove references to the choco installer from the guides and revert to manually copying the binary into `%PATH%`.

**Comment by Tonkpils** [comment]
Once the patch for upgrade is merged this shouldn't be an issue since it would fix self update for windows though not sure what the limitations would be of installing the CLI with the choco package and upgrading it with the self-update command would be.

**Comment by NobbZ** [comment]
Well, the usual things might happen that happen when actual binary and package manager get out of sync. I'm not sure in which category choco belongs, but those are situations I have seen already:

* Downgrade the out-of-sync executable on every system update to get back in sync
* Remove the out-of-sync executable and ring all bells and yell the word "infected" at the user
* Yell the word "infected" at the user without doing anything else
* Assume the user knows what he is doing and just warn about some wrong checksum
* Do not even realise that we are out of sync until the package gets updated and overwrite the altered binary without any warnings

Because of this possibilities and even others, it is considered bad style to alter anything that has been created by a package manager.

If though the patch actually doesn't alter the choco managed binary, but a per-user-local-install in his/her home-folder, then this seems to be fine, but might lead into further trouble depending on the order of pathes in `$PATH`/`%PATH%`. Hard to debug for people that do not know anything about how their system finds executables...

Ergo: do not touch or shadow a package managed installation unless you do tell the user about the consequences before that and ask if he really wants to get into the possible troubles.

**Comment by rpottsoh** [comment]
`exercism upgrade` works now.  Should this issue be closed?

**Comment by Tonkpils** [comment]
Yup! Thanks for pointing it out
